plugins/chain_api_plugin/chain_api_plugin.cpp

```diff
@@ -41,6 +41,48 @@ parse_params<chain_apis::read_only::get_transaction_status_params, http_params_t
    }
 }
 
+// if actions.data & actions.hex_data provided, use the hex_data since only currently support unexploded data
+template<>
+chain_apis::read_only::get_transaction_id_params
+parse_params<chain_apis::read_only::get_transaction_id_params, http_params_types::params_required>(const std::string& body) {
+   if (body.empty()) {
+      EOS_THROW(chain::invalid_http_request, "A Request body is required");
+   }
```

Comment:
Is there a reason why this is not in the `try` block below?

Reply:
Mainly to follow similar pattern of parse_params functions in chain_api_pplugin, http_plugin where check for empty body done like that, etc and it's not going to throw

Reply:
OK, it is just that the other `EOS_THROW` below would be caught by the `EOS_RETHROW_EXCEPTIONS` and annotated with `Invalid transaction`, but not this first one. If this is what we want then I'm fine with that.

Reply:
They should be still similar:
`
curl -X POST http://127.0.0.1:8888/v1/chain/get_transaction_id --header 'content-type: application/json' --data '{}'

{"code":400,"message":"Invalid Request","error":{"code":3200006,"name":"invalid_http_request","what":"invalid http request","details":[{"message":"Transaction actions are missing or invalid","file":"chain_api_plugin.cpp","line_number":79,"method":"parse_params"}]}}

curl -X POST http://127.0.0.1:8888/v1/chain/get_transaction_id --header 'content-type: application/json' --data ''

{"code":400,"message":"Invalid Request","error":{"code":3200006,"name":"invalid_http_request","what":"invalid http request","details":[{"message":"A Request body is required","file":"chain_api_plugin.cpp","line_number":49,"method":"parse_params"}]}}
`

Reply:
OK, looks good, I though the message from `EOS_RETHROW_EXCEPTIONS` would be added, but it doesn't seem to be the case.